web-app/django/VIM/apps/instruments/views/instrument_detail.py

```diff
@@ -0,0 +1,45 @@
+from django.views.generic import DetailView
+from VIM.apps.instruments.models import Instrument, Language
+
+
+class InstrumentDetail(DetailView):
+    """
+    Displays details of a specific instrument.
+    """
+
+    model = Instrument
+    template_name = "instruments/detail.html"
+    context_object_name = "instrument"
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        instrument = self.get_object()
+
+        context["wikidata_id"] = instrument.wikidata_id
+        context["default_image_url"] = (
+            instrument.default_image.url if instrument.default_image else None
+        )
+        context["hornbostel_sachs_class"] = instrument.hornbostel_sachs_class
+        context["mimo_class"] = instrument.mimo_class
+
+        # Get english name
+        context["instrument_name_en"] = instrument.instrumentname_set.get(
+            language__en_label="english"
+        ).name
+
+        # Get all instrument names in all languages
+        instrument_names = []
+        for instrument_name in instrument.instrumentname_set.all():
+            instrument_names.append(
+                {
+                    "name": instrument_name.name,
+                    "source_name": instrument_name.source_name,
+                    "language_code": instrument_name.language.wikidata_code,
+                    "language_id": instrument_name.language.wikidata_id,
+                    "language_en_label": instrument_name.language.en_label,
+                    "language_autonym": instrument_name.language.autonym,
+                }
+            )

```

Comment:
Ok, this is the place where I really do think you'd want to go ahead and optimize. Every step in the loop, this means you'll have a separate SQL query, which means if we end up having a hundred names for the instrument, we'll have a hundred queries. 

On line 32, we'll do one query to get all of the instrument names for the instrument, and then starting on line 37, we'll do another query to get the language info for each name.

If you use the [`.select_related` method](https://docs.djangoproject.com/en/5.0/ref/models/querysets/#select-related), you can get this all in one query (essentially, you are telling Django to make a query with a join):
`instrument.instrumentname_set.all().select_related("language")`